### PR TITLE
Auto deletion of old backup/swap/undo files

### PR DESCRIPTION
**Pull request by matt1003**
See individual commits for details.

Please note that I have no yet had a chance to do much testing on these changes.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,23 +2,43 @@
 
 **Centralize Backup, Swap, Undo**
 
-Create directories to store unique backup, swap, and undo files in a
-centralized place.
+Create and manage directories to store unique backup, swap, and undo
```
Comment (her) [inline README.md:5]: Nice wording here 👍

File: README.md [hunk 2]
```diff
@@ -2,23 +2,43 @@
...
+files in a centralized place.
 
-Backup, swap, and undo are created in `~/.vim` 
+Backup, swap, and undo are created in `$VIMHOME` (which defaults to
+`~/.vim` on Unix based systems).
 
 ```sh
-.vim
+$VIMHOME
```
Comment (her) [inline README.md:12]: Fantastic. 🎉

File: plugin/central.vim [hunk 3]
```diff
@@ -3,16 +3,43 @@
 " Version: 0.2.0
 " License: BSD
 
-set backupdir=~/.vim/backup//
-set directory=~/.vim/swap//
-set undodir=~/.vim/undo//
+set backupdir=$VIMHOME/backup//
```
Comment (her) [inline plugin/central.vim:7]: Big improvement here. This really makes the plugin much friendlier to other OS

File: plugin/central.vim [hunk 4]
```diff
@@ -3,16 +3,43 @@
...
 
-if !isdirectory($HOME . "/.vim/backup")
-    call mkdir($HOME . "/.vim/backup", "p")
+for dir in [ &backupdir, &directory, &undodir ]
+    if !isdirectory(dir)
+        call mkdir(dir, 'p')
+    endif
+endfor
+
+set backup swapfile undofile
```
Comment (her) [inline plugin/central.vim:17]: may as well 😁 👏

File: plugin/central.vim [hunk 5]
```diff
@@ -3,16 +3,43 @@
...
-set undodir=~/.vim/undo//
+set backupdir=$VIMHOME/backup//
+set directory=$VIMHOME/swap//
+set undodir=$VIMHOME/undo//
 
-if !isdirectory($HOME . "/.vim/backup")
-    call mkdir($HOME . "/.vim/backup", "p")
+for dir in [ &backupdir, &directory, &undodir ]
+    if !isdirectory(dir)
+        call mkdir(dir, 'p')
```
Comment (her) [inline plugin/central.vim:13]: so here with 

`call mkdir(dir, 'p')` This suppresses any error messages where the directory may already exist, right? 

Awesome stuff, Matt.

File: plugin/central.vim [hunk 6]
```diff
@@ -3,16 +3,43 @@
...
+    endif
+endfor
+
+set backup swapfile undofile
+
+if !exists('g:central_cleanup_enable')
+    let g:central_cleanup_enable = 30 "days
+endif
+
+if g:central_cleanup_enable > 0
```
Comment (her) [inline plugin/central.vim:23]: Love the configuration option!

File: plugin/central.vim [hunk 7]
```diff
@@ -3,16 +3,43 @@
...
+    let epoch = localtime() - (g:central_cleanup_enable * 86400)
+    for dir in [ &backupdir, &directory, &undodir ]
+        for file in split(glob(substitute(dir, '//', '/*', '')))
+            if getftime(file) < epoch
```
Comment (her) [inline plugin/central.vim:27]: Neat. had no idea vim script had this function. That's handy :)

File: plugin/central.vim [hunk 8]
```diff
@@ -3,16 +3,43 @@
...
-if !isdirectory($HOME . "/.vim/swap")
-    call mkdir($HOME . "/.vim/swap", "p")
+
+if !exists('g:central_multiple_backup_enable')
+    let g:central_multiple_backup_enable = 1
 endif
-if !isdirectory($HOME . "/.vim/undo")
-    call mkdir($HOME . "/.vim/undo", "p")
+
+if g:central_multiple_backup_enable == 1
```
Comment (her) [inline plugin/central.vim:38]: This is actually a life saver.

I've had cases where, I thought I checked something into git, hadn't, checked that file out, and it was blown away :( 

Sweet addition ❤️

**Review by her — APPROVED** [review]
@matt1003 you're the greatest first contributor ever! Thanks for making such a great contribution :)

**Comment by her** [comment]
@matt1003 this is great. If you're up for it, I added you as a collaborator so you can merge this PR, if you want? Would be great if you stuck around! 

Otherwise am happy to merge it 👍

**Comment by her** [comment]
oooo 

Also, mind adding your name to the 'author(s)' section on the doc?

File: plugin/central.vim (hunk 5 above)
Comment (matt1003) [reply plugin/central.vim:13]: The `call mkdir(dir, 'p')` will **not** suppress errors if the directory already exists; however the `if !isdirectory(dir)` should render this moot.

File: plugin/central.vim (hunk 8 above)
Comment (matt1003) [reply plugin/central.vim:38]: Last week I managed to overwrite two days worth of work. By executing `:w` multiple times I lost my backup, and I also managed to screw up my undo history (not sure how I did the latter). Anyway, this is what prompted me to play around with multiple backups. The auto clean up feature was then a consequence of making sure the multiple backups did not get out of control.

**Comment by matt1003** [comment]
@her, I am more than happy to be a collaborator. I will merge this pull request an assume responsibility if it creates any bugs.

**Comment by her** [comment]
Great stuff 🍻

File: plugin/central.vim (hunk 5 above)
Comment (her) [reply plugin/central.vim:13]: ahhh, very nice! thanks for explaining that

File: plugin/central.vim (hunk 8 above)
Comment (her) [reply plugin/central.vim:38]: 😭 owie that bites! 

>The auto clean up feature was then a consequence of making sure the multiple backups did not get out of control.

lol i guess it's true what they say then, 'necessity is the mother of invention' 😄

**Pull request merged**